modules/order/src/Plugin/Field/FieldFormatter/OrderItemTable.php

```diff
@@ -23,14 +23,19 @@ class OrderItemTable extends FormatterBase {
    * {@inheritdoc}
    */
   public function viewElements(FieldItemListInterface $items, $langcode) {
-    $order = $items->getEntity();
-    return [
-      '#type' => 'view',
-      // @todo Allow the view to be configurable.
-      '#name' => 'commerce_order_item_table',
-      '#arguments' => [$order->id()],
-      '#embed' => TRUE,
-    ];
+    $element = [];
+    if ($items->count() > 0) {
```

Comment:
I see it as related. If the items are empty (count() == 0), then do not even bother inserting a view. Otherwise insert 1 single view at 0 delta.

Basically, if some $items are empty then we shouldn't return anything at all in $element[0]. Otherwise it may appear misleading - empty items do get formatted into some renderable array.

I am thinking right now.. Since I do not know well inner kitchen of commerce 8.x maybe the view "commerce_order_item_table" has some "empty" logic?

Well, again, I made what I deemed the best given my limited knowledge about the context and have no preference here on whether to keep that count(). :)